src/main/java/redis/clients/jedis/JedisClusterInfoCache.java

```diff
@@ -146,53 +159,27 @@ private void discoverClusterSlots(Jedis jedis) {
   }
 
   private HostAndPort generateHostAndPort(List<Object> hostInfos) {
-    return new HostAndPort(SafeEncoder.encode((byte[]) hostInfos.get(0)),
-        ((Long) hostInfos.get(1)).intValue());
-  }
-
-  public JedisPool setupNodeIfNotExist(HostAndPort node) {
-    w.lock();
-    try {
-      String nodeKey = getNodeKey(node);
-      JedisPool existingPool = nodes.get(nodeKey);
-      if (existingPool != null) return existingPool;
-
-      JedisPool nodePool = new JedisPool(poolConfig, node.getHost(), node.getPort(),
-          connectionTimeout, soTimeout, password, 0, clientName, false, null, null, null);
-      nodes.put(nodeKey, nodePool);
-      return nodePool;
-    } finally {
-      w.unlock();
-    }
-  }
-
-  public JedisPool setupNodeIfNotExist(HostAndPort node, boolean ssl) {
```

Comment:
This breaks backward compatibility. But I don't see any usage either. I guess we should have separate PRs to remove/deprecate this.